### PR TITLE
Fix FP on S4335 when interfaces have computed properties

### LINKED ISSUE #2459: Fix FP on S4335 when interfaces have computed properties
Community: https://community.sonarsource.com/t/ts-interface-dynamic-property-names-cause-a-false-positive-report-of-rspec-4335/36832

--
After a quick investigation, the function helper `isEmptyInterface` considers interfaces with only computed properties as being empty. The implementation should use the declarations of the type symbol to check the emptiness of the interface, i.e. the absence of members.

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #2459

### CODE REVIEW

File: eslint-bridge/src/rules/no-useless-intersection.ts [hunk 1]
```diff
@@ -75,12 +75,20 @@ function isNullLike(tp: ts.Type): boolean {
...
+  );
 }
 
-function isStandaloneInterface({ declarations }: ts.Symbol) {
+function isStandaloneInterface(typeSymbol: ts.Symbol) {
   // there is no declarations for `{}`
   // otherwise check that none of declarations has a heritage clause (`extends X` or `implements X`)
+  if (!typeSymbol) {
+    return false;
+  }
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/src/rules/no-useless-intersection.ts:90]: I am not sure I understand why there is this if-statement here. `typeSymbol` is supposed to be well-typed so it should never be undefined, right ? Is it an oversight or did you encounter some failures while running the ruling its that you are not able to reproduce in a unit test ?

Comment (saberduck) [reply eslint-bridge/src/rules/no-useless-intersection.ts:90]: it's the latter, there were failures in ruling - and I was not able to extract them

Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/rules/no-useless-intersection.ts:90]: Okay, then can you maybe add a little comment about this before the return statement ?

**Review by yassin-kammoun-sonarsource — APPROVED** [review]

**Comment by saberduck** [comment]
merged with red QG, because missing coverage is related to missing type information

**Pull request merged**